src/main/kotlin/io/github/moulberry/notenoughupdates/util/SkyBlockTime.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2023 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.util
+
+class SkyBlockTime constructor(
+    var year: Int = 1,
+    var month: Int = 1,
+    var day: Int = 1,
+    var hour: Int = 0,
+    var minute: Int = 0,
+    var second: Int = 0,
+) {
```

Comment:
Consider using a data class:
```suggestion
data class SkyBlockTime(
    val year: Int = 1,
    val month: Int = 1,
    val day: Int = 1,
    val hour: Int = 0,
    val minute: Int = 0,
    val second: Int = 0,
) {
```
This auto generates a `toString()` method as well as a `copy()` method (in place of your `duplicate()` method. In addition making the fields `val` still allows you to change the fields by using `.copy(hour = 0, minutes = 0  /* etc */)` with all the benefits of immutability.
(Data classes can have either `val` or `var` fields, if you prefer having a mutable data class)

Reply:
I wanted to make the primary constructor private. That's the intention behind this. Additionally, the fields have to be mutable for the logic I want to use them for. Similar to Calendar

Reply:
implemented.